frontend/components/NavigationBar.js

```diff
@@ -0,0 +1,380 @@
+    let mediaDropdown = document.getElementById("mediaDropdown");
+    aboutUsDropdown.addEventListener("mouseover", changeAboutUsToArrowDown);
+    aboutUsDropdown.addEventListener("mouseleave", changeAboutUsToArrowRight);
+    mediaDropdown.addEventListener("mouseover", changeMediaToArrowDown);
+    mediaDropdown.addEventListener("mouseleave", changeMediaToArrowRight);
+  }, []);
+
+  const checkWindowWidth = () => {
+    const checkWidth = window.innerWidth < "950";
+    setCompactNavBar(checkWidth);
+  };
+
+  const checkBottomScreen = () => {
+    const checkBottom =
+      Math.ceil(window.innerHeight + window.scrollY) >=
+      document.documentElement.scrollHeight;
+    setHiddenNavBar(checkBottom);
+  };
+
+  const checkTopScreen = () => {
+    const checkTop = window.scrollY < 60;
+    setClearNavBar(checkTop);
+  };
+
+  const changeAboutUsToArrowDown = () => {
+    let aboutUsText = document.getElementById("aboutUsText");
+    aboutUsText.innerHTML = "About Us ▾";
+  };
+
+  const changeAboutUsToArrowRight = () => {
+    let aboutUsText = document.getElementById("aboutUsText");
+    aboutUsText.innerHTML = "About Us ▸";
+  };
+
+  const changeMediaToArrowDown = () => {
+    let mediaText = document.getElementById("mediaText");
+    mediaText.innerHTML = "Media ▾";
+  };
+
+  const changeMediaToArrowRight = () => {
+    let mediaText = document.getElementById("mediaText");
+    mediaText.innerHTML = "Media ▸";
+  };
+
+  /* return hamburger navbar if screen size is smaller than 950px,
+     otherwise return regular navbar */
+  return compactNavBar ? (
+    <CompactNavigationBar />
+  ) : (
+    /* return clear or hidden navbar if at top or bottom of screen respectively,
+       otherwise return regular navbar */
+    <nav
+      className={
+        clearNavBar && router.pathname != "/404"
+          ? `${styles.clearNavBar} ${styles.navBarContainer}`
+          : hiddenNavBar
+          ? `${styles.hiddenNavBar} ${styles.navBarContainer}`
+          : styles.navBarContainer
+      }
+    >
+      <div className={styles.navBarGrid}>
+        <div className={styles.logoContainer}>
+          <Link href="/">
+            <a>
+              {/* change WIT logo color depending on 
+                  the device theme and current scroll position */}
+              <Image
+                className={styles.logoGridItem}
+                src={
+                  darkMode || (clearNavBar && router.pathname != "/404")
+                    ? "/logo-white.png"
+                    : "/logo-black.png"
+                }
+                alt="wit logo"
+                width={40}
+                height={35}
+              />
+            </a>
+          </Link>
+        </div>
+        <div className={styles.emptyGridItem}></div>
+        <div className={`${styles.homeGridItem} ${styles.linkContainer}`}>
+          <Link href="/">
+            <div className={styles.linkContent}>
+              <a>Home</a>
```

Comment:
Use the `map` function for routes/headings that aren't included in the about us/media dropdowns too.